chore(docs): adding Tracetest integration on Keptn (#196)
keptn-contrib/artifacthub: tracetest/1.0.0/README.md

By using the Keptn [Job Executor Service](https://github.com/keptn-contrib/job-executor-service) plugin, we can upload a Tracetest test definition and a CLI configuration to a service and run a test using the following job:

```yaml
apiVersion: v2
actions:
- name: "Run tracetest on your service"
  events:
    - name: "sh.keptn.event.test.triggered"
  tasks:
    - name: "Run tracetest"
      files:
        - data/test-definition.yaml
        - data/tracetest-cli-config.yaml
      image: "kubeshop/tracetest:latest"
      cmd:
        - tracetest
      args:
        - --config
        - /keptn/data/tracetest-cli-config.yaml
        - test
        - run
        - --definition
        - /keptn/data/test-definition.yaml
        - --wait-for-result

```

## Quickstart

Here we will show how to use Tracetest to run these tests and help in your delivery and testing workflows using the [Pokeshop](https://docs.tracetest.io/pokeshop/) example, available at `http://demo-pokemon-api.demo` in a Kubernetes cluster.

### Prerequisites

In your Kubernetes cluster you should have:

> 1. `Keptn 1.0.x` [installed](https://keptn.sh/docs/1.0.x/install/)

Reviewer: We have now released Keptn 1.1.0.  It would be nice if Tracetest worked with that and I think it should.

Author: It works! I'm updating this doc, thanks!

2. `Job Executor Service 0.3.x` [installed](https://github.com/keptn-contrib/job-executor-service/blob/main/docs/INSTALL.md)
3. `Tracetest` server [installed](https://docs.tracetest.io/deployment/kubernetes) on `tracetest` namespace 

On your machine you should have:

1. `Keptn CLI` [installed](https://keptn.sh/docs/1.0.x/install/cli-install/)
2. and already [authenticated](https://keptn.sh/docs/1.0.x/install/authenticate-cli-bridge/) with Keptn API.

With everything set up, we will start configuring Keptn and Tracetest.

> ### 1. Setup a project and a service.

Reviewer: ```suggestion
### 1. Set up a project and a service.
```
The verb is two words -- "setup" as one word is a noun or adjective.

 
Keptn works with [concepts](https://keptn.sh/docs/concepts/glossary/) of a Project (element to maintain multiple services forming an application in stages) and a Service (smallest deployable unit and is deployed in all project stages according to the order).

Usually, these resources are managed by Keptn during Sequences (a set of tasks for realizing a delivery or operations process). In this example, we will create a sequence and run a Tracetest test in a system, once the sequence is triggered.
 
To do that, we will do the following steps:

1. Create a skeletal [`shipyard.yaml`](https://github.com/kubeshop/tracetest/tree/main/examples/keptn-integration/shipyard.yaml) file with the following content:
```yaml
apiVersion: "spec.keptn.sh/0.2.2"
kind: "Shipyard"
metadata:
 name: "shipyard-keptn-tracetest-integration"
spec:
 stages:
   - name: "production"
     sequences:
       - name: "test-sequence"
         tasks:
           - name: "test-services"
```

2. Create a new `keptn-tracetest-integration` project using that `shipyard` file:
```sh
keptn create project keptn-tracetest-integration -y -s shipyard.yaml
```
 

> **Note:** Keptn may ask you to have a Git repository for this project to enable GitOps. If so, you need to create an empty Git repository and a Git token and pass it through the flags `--git-remote-url`, `-`-git-user`, and `--`git-token`. More details about this setup can be seen on [Keptn docs/Git-based upstream](https://keptn.sh/docs/1.0.x/manage/git_upstream).

Reviewer: ```suggestion
**Note:** Keptn requires you to have a Git repository for this project to enable GitOps. Before you run this command, you need to create an empty Git repository and a Git token and pass it through the flags `--git-remote-url`, `-`-git-user`, and `--`git-token`. More details about this setup can be seen on [Keptn docs/Git-based upstream](https://keptn.sh/docs/1.0.x/manage/git_upstream).
```
For Keptn 1.y.z, the upstream git repo is required, period -- no possibility that you can create a project without it.

4. Create [`job-config.yaml`](https://github.com/kubeshop/tracetest/tree/main/examples/keptn-integration/job-config.yaml):
```yaml
apiVersion: v2
actions:
- name: "Run tracetest on your service"
  events:
    - name: "sh.keptn.event.test-services.triggered"
  tasks:
    - name: "Run tracetest"
      files:
        - data/test-definition.yaml
        - data/tracetest-cli-config.yaml
      image: "kubeshop/tracetest:latest"
      cmd:
        - tracetest
      args:
        - --config
        - /keptn/data/tracetest-cli-config.yaml
        - test
        - run
        - --definition
        - /keptn/data/test-definition.yaml
        - --wait-for-result
```

This job will run Tracetest every time an event of `test-services` happens, listening to the event `sh.keptn.event.test-services.triggered` (event emitted by the `test-services` task on the `test-sequence` sequence).

5. Add this job as a resource on Keptn:
```sh
keptn add-resource --project keptn-tracetest-integration --service pokeshop --stage production --resource job-config.yaml --resourceUri job/config.yaml
```
 

> ### 3. Setup Job Executor Service to see events emitted by the test step.

Reviewer: ```suggestion
### 3. Set up Job Executor Service to see events emitted by the test step.
```
